src/main/java/org/geysermc/discordbot/commands/moderation/ForumCommand.java

```diff
@@ -0,0 +1,304 @@
+            String issue = event.optString("issue", "");
+
+            if (!targetUser.isEmpty()) {
+                issue += " " + targetUser;
+            }
+
+            ForumChannel forumChannel = event.getGuild().getForumChannelById(FORUM_CHANNEL_ID);
+            if (forumChannel == null) {
+                event.reply("Forum channel not found.").queue();
+                return;
+            }
+
+            forumChannel.createForumPost(title, MessageCreateData.fromContent(issue))
+                    .queue(
+                            unused -> event.reply("Post is created").queue(),
+                            error -> event.reply("Could not create post").queue()
+                    );
+        }
+    }
+
+    public static class CloseOldPostSubCommand extends SlashCommand {
+        public CloseOldPostSubCommand() {
+            this.name = "close-old";
+            this.help = "Close old posts";
+            this.userPermissions = new Permission[]{Permission.MESSAGE_MANAGE};
+            this.options = List.of(new OptionData(OptionType.INTEGER, "days", "The minimum age in days of posts that will be closed in bulk", true));
+        }
+
+        @Override
+        protected void execute(@NotNull SlashCommandEvent event) {
+            int days = Objects.requireNonNull(event.getOption("days")).getAsInt();
+            Checks.notNull(event.getGuild(), "server");
+            Guild guild = event.getGuild();
+
+            ForumChannel forumChannel = guild.getForumChannelById(FORUM_CHANNEL_ID);
+            if (forumChannel == null) {
+                event.reply("Forum channel not found.").queue();
+                return;
+            }
+
+            for (ThreadChannel channel : forumChannel.getThreadChannels()) {
+                if (channel.isArchived() || channel.isLocked() || channel.isPinned()) {
+                    continue;
+                }
+                OffsetDateTime channelCreated = channel.getTimeCreated();
+                OffsetDateTime closeTime = OffsetDateTime.now().minusDays(days);
+                if (channelCreated.isBefore(closeTime)) {
+                    ThreadChannelManager manager = channel.getManager().setArchived(true);
+                    manager.queue();
+                }
+            }
+
+            event.reply("All forum posts older than " + days + " days have been closed!").queue();
+        }
+    }
+
+    public static class ClosePostSubCommand extends SlashCommand {
+        public ClosePostSubCommand() {
+            this.name = "close";
+            this.help = "Close post";
+            this.userPermissions = new Permission[]{Permission.CREATE_PUBLIC_THREADS};
+        }
+
+        @Override
+        protected void execute(@NotNull SlashCommandEvent event) {
+            if (!(event.getChannel() instanceof ThreadChannel) || !event.getChannel().asThreadChannel().getParentChannel().getId().equals(FORUM_CHANNEL_ID)) {
```

Comment:
What is the point of `event.getChannel().asThreadChannel().getParentChannel().getId().equals(FORUM_CHANNEL_ID)`?

Reply:
Just changed that bit, was basically to make sure we are doing to command in the correct forumpost, retrieving the channel id as normal will result in the id of that one post but in this case we need the id of the parent ( forum channel itself )